pkg/resourcecreator/serviceaccount/serviceaccount.go

```diff
@@ -21,5 +21,11 @@ func Create(source resource.Source, ast *resource.Ast, options resource.Options)
 		ObjectMeta: objectMeta,
 	}
 
+	if options.SkattUsePullSecret {
```

Comment:
This is probably not needed anymore since nais deployment generator will automatically add these to the deployment directly.